app/views/institutions/_form.haml

```diff
@@ -37,7 +36,8 @@
       .col.px-1
         = f.label :name, :class => 'block'
       .col
-        = f.text_field :name, readonly: @readonly, :class => 'input-block'
+        = f.text_field :name, readonly: @readonly, :class => ['input-block', @institution.errors.has_key?(:name) ? 'input-required':'']
+        = field_errors (@institution.errors.full_messages_for(:name))
```

Comment:
I think retrieving the actual error messages could be deferred to the helper method.
```suggestion
        = field_errors f, :name
```

So `field_errors` receives the form builder and the name of the field, and retrieves the messages on its own.

Reply:
No, I meant `f`, the form context. It contains a reference to the object (`f.object`). This avoids some duplication or far-away contextual matching to the form object, which is just easily accessible through the builder.